R/logging.R

```diff
@@ -0,0 +1,110 @@
+#' Get node log files from Azure Storage. By default, this operation will print the files on screen.
+#'
+#' @param cluster The cluster object
+#' @param nodeId Id of the node
+#' @param type The type of logs the user wants: "stdout" or "stderr"
+#' @param verbose Flag for printing log files onto console
+#'
+#' @param ... Further named parameters
+#' \itemize{
+#'  \item{"localPath"}: { Path to save file to }
+#'  \item{"overwrite"}: { Will only overwrite existing localPath }
+#'}
+#'
+#' @examples
+#' stdoutText <- getClusterLogs(cluster, "tvm-1170471534_1-20170829t072146z", type = "stdout", verbose = FALSE)
+#' getClusterLogs(cluster, "tvm-1170471534_2-20170829t072146z", type = "stderr", localPath = "abc.txt", overwrite = TRUE)
+#' @export
+getClusterLogs <-
+  function(cluster,
+           nodeId,
+           type = c("stdout", "stderr"),
```

Comment:
Is this pattern going to limit us in the future? I.e. having file 'types'. It seems a bit contrived on our end to dictate what a file type is... Perhaps the app has it's own logging (just guessing) that doesn't go to stdout? In that case, shouldn't we just accept a file name (e.g. stdout.txt or stderr.txt) and try to find that file on the node? All of file names and other relevate info can simply go into the docs.

Reply:
i think we shouldn't just accept a filename... because r users don't normally specify a separate file for logs - they just look at whatever is printed into the R console (which i __believe__ is stdout - but we should verify... whatever is printed out to the console is what we should be sending back)

I think __optionally__, users should be able to pass in a specified file name that they may be writing logs out to

Reply:
Well... yes, that is the main scenario, but not the only scenario. If we allow 'anything' then we open up the doors to all sorts of scenarios and unblock the user from doing whatever they want. If we manage categories ourselves then we can very much limit the experience and probably cause a breaking api change in the future.

If we really just care about logs, then we shouldn't take a 'type' at all.. we should just get all the logs.